### PR TITLE
Test Suite + Model Tests

### PR DESCRIPTION
**Pull request by paultannenbaum**
- Fixes broken test runner
- All model relationships tested. Other functionality marked as pending
- Reorganized all model files to list properties in this order: relationships, properties, methods, observable methods

### CODE REVIEW

File: app/pods/bucket-type/model.js [hunk 1]
```diff
@@ -28,6 +28,23 @@ var BucketType = DS.Model.extend({
     bucketList: DS.belongsTo('bucket-list'),
 
     /**
+     * Bucket Type Properties object.
```
Comment (dmitrizagidulin) [inline app/pods/bucket-type/model.js]: So, out of sheer curiosity, why change the order of the properties? It was in alphabetical order before - what's the logic now? :)

**Comment by dmitrizagidulin** [comment]
Aside from the question about reordering of all those properties, my only other comment is -- don't forget to add the test running instructions (cli and browser) to the README. 
Looks great otherwise.

File: app/pods/bucket-type/model.js (hunk 1 above)
Comment (paultannenbaum) [inline app/pods/bucket-type/model.js]: I mentioned it the bullet points, but I reordered all models so relationships are first, attributes second, methods third, and observable methods last. I personally think its easier to scan a model that way. I probably undid some of the alphabetizing when I did that, so I can at least make sure they are alphabetized within their given order. I'll go ahead and do that before merging.

**Comment by paultannenbaum** [comment]
Forgot about the README instructions, thanks! Will add and then merge.

File: app/pods/bucket-type/model.js (hunk 1 above)
Comment (dmitrizagidulin) [inline app/pods/bucket-type/model.js]: Ah, I missed that somehow. That's cool, then.

**Comment by dmitrizagidulin** [comment]
Sounds good, +1

**Pull request merged**